libvast/src/concept/parseable/vast/legacy_type.cpp

```diff
@@ -64,7 +64,8 @@ bool legacy_type_parser::parse(Iterator& f, const Iterator& l,
     | "duration"_p  ->* type_factory<legacy_duration_type>
     | "time"_p      ->* type_factory<legacy_time_type>
     | "string"_p    ->* type_factory<legacy_string_type>
-    | "pattern"_p   ->* type_factory<legacy_pattern_type>
+    // We removed support for pattern types with VAST v3.0.
+    // | "pattern"_p   ->* type_factory<legacy_pattern_type>
```

Comment:
Why not ditch it entirely?

Reply:
I cannot remove the `legacy_pattern_type` as that also changes how some other legacy types are serialized, so I figured it's best to just leave it in and comment it out here. Once we upgrade the minimum partition version to 3 (I believe) we can get rid of it. Patryk removed the last remaining serialized uses of it with the CAF upgrade.

I would really love to get rid of the legacy type system, but we're still not quite there yet.